src/uu/free/src/free.rs

```diff
@@ -433,18 +433,30 @@ fn humanized(kib: u64, si: bool) -> String {
 }
 
 fn detect_unit(arg: &ArgMatches) -> fn(u64) -> u64 {
+    let si = arg.get_flag("si");
     match arg {
         _ if arg.get_flag("bytes") => |kib: u64| ByteSize::kib(kib).0,
-        _ if arg.get_flag("kilo") => |kib: u64| ByteSize::kib(kib).0 / KB,
-        _ if arg.get_flag("mega") => |kib: u64| ByteSize::kib(kib).0 / MB,
-        _ if arg.get_flag("giga") => |kib: u64| ByteSize::kib(kib).0 / GB,
-        _ if arg.get_flag("tera") => |kib: u64| ByteSize::kib(kib).0 / TB,
-        _ if arg.get_flag("peta") => |kib: u64| ByteSize::kib(kib).0 / PB,
+        _ if arg.get_flag("kilo") || (!si && arg.get_flag("kibi")) => {
```

Comment:
While the code works fine, the `!si` is very confusing ;-)

My suggestion is to change the definition of `--si` from `ArgAction::SetFalse` to `ArgAction::SetTrue` and then use `si`. I don't know why `ArgAction::SetFalse` was used, it seems incorrect to me.

Reply:
I'm also confused. I'm guessing it has something to do with the `bytesize::ByteSize::to_string_as()` function when `--human` flag is set. This function accepts an `si_unit: bool` argument, which is different from the `--si` option in free. So I will modify all judgments of `si` flag.